### PR TITLE
Implement session state

### LINKED ISSUE #16: Implement session state and OnSessionStateChanged equivalent in js
Please refer to https://github.com/authgear/authgear-sdk-android/issues/9 for the rationale.

### PR DESCRIPTION
**Pull request by axolotl0212**
closes #16 

not tested yet

### CODE REVIEW

**Comment by louischan-oursky** [comment]
@kiootic In case you have missed this PR.

File: packages/authgear-core/src/container.ts [hunk 1]
```diff
@@ -89,6 +92,16 @@ export abstract class BaseContainer<T extends BaseAPIClient> {
    */
   expireAt?: Date;
 
+  /**
+   * @internal
+   */
+  sessionState: SessionState;
```
Comment (kiootic) [inline packages/authgear-core/src/container.ts:98]: Both the session state value and the listeners are suppsoed to be used by developers, so we should make them public.

File: packages/authgear-core/src/types.ts [hunk 2]
```diff
@@ -223,3 +226,70 @@ export interface _OIDCTokenResponse {
...
+ * state at any given point in time.
+ *
+ * For example, the session state of an authgear instance that is just constructed always has
+ * "Unknown". After a call to [Authgear.configure], the session state would be
+ * "LoggedIn" if a previous session is found, or "NoSession" if there is
+ * no such sessions.
+ *
+ * @public
+ */
+export type SessionState = "LoggedIn" | "NoSession" | "Unknown";
```
Comment (kiootic) [inline packages/authgear-core/src/types.ts:243]: I suppose using UPPER_SNAKE_CASE matches the JS convention on constant values better.

**Comment by axolotl0212** [comment]
@kiootic Updated and rebased

File: packages/authgear-core/src/container.ts [hunk 3]
```diff
@@ -69,6 +72,16 @@ export abstract class BaseContainer<T extends BaseAPIClient> {
...
 
+  /**
+   * @public
+   */
+  sessionState: SessionState;
+
+  /**
+   * @public
+   */
+  onSessionStateChangedListeners: OnSessionStateChangedListener[];
```
Comment (kiootic) [inline packages/authgear-core/src/container.ts:83]: This should be internal. Instead, expose `_addOnSessionStateChangedListener` and `_removeOnSessionStateChangedListener` for public consumption.

**Review by kiootic — APPROVED** [review]

**Pull request merged**